### DISCUSSION TITLE
Uploading file with With USB Copy?

### DISCUSSION THREAD

**Discussion by stevecolt** [earlephilhower/arduino-pico#1270] [Q&A] [answered]
Hi,
Is there a way to upload the arduino generated code to a blank Pico using a simple USB copy/paste command? I don't want to go through the IDE every time I build a new board, and would be useful for field uploads to non-programmers.
Would renaming the xxxx.ino.uf2 file created in the temp/build folder, to NEW.UF2, then go into bootloader and paste work?
If not what's the solution, and could it be relatively easy to include the way of saving the appropriate file?
I've seen a similar post on the Arduino forum that suggested you could using some python tools, but it would be much nicer if you could just download a created file.
Many thanks

Steve

**Comment by earlephilhower** [comment, marked as answer]
>  Would renaming the xxxx.ino.uf2 file created in the temp/build folder, to NEW.UF2, then go into bootloader and paste work?

No need to rename the file, just copy it as-is and it will work.  The filename isn't used by the bootrom, only the data inside.  You can copy it in Windows Explorer and it will "just work".

Note that, last I checked, Apple broke the file copy to upload UF2 behavior.  If you're on a Mac and plain copying doesn't work, please check the Raspberry Pi forums/blogs for a solution.

**Reply by stevecolt** [reply]
Excellent, thank you.
I'm still trying to figure out how UF2 works on Windows. 
Steve

**Reply by earlephilhower** [reply]
Just hold BOOTSEL while plugging in the Pico, then drag-n-drop the XXX.UF2 onto the drive and it should flash and reboot.  Windows and Linux, it's pretty painless.
